flake.nix

```diff
@@ -0,0 +1,83 @@
+{
+  description = "A library for probabilistic programming in Haskell.";
+
+  nixConfig = {
+    extra-substituters = [
+      "https://tweag-monad-bayes.cachix.org"
+    ];
+    extra-trusted-public-keys = [
+      "tweag-monad-bayes.cachix.org-1:tmmTZ+WvtUMpYWD4LAkfSuNKqSuJyL3N8ZVm/qYtqdc="
+    ];
+  };
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-unstable";
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
+    flake-utils.url = "github:numtide/flake-utils";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+    pre-commit-hooks.inputs.nixpkgs.follows = "nixpkgs";
+    pre-commit-hooks.inputs.flake-utils.follows = "flake-utils";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-compat,
+    flake-utils,
+    pre-commit-hooks,
+  }:
+    flake-utils.lib.eachSystem
+    [
+      flake-utils.lib.system.x86_64-linux
+      flake-utils.lib.system.aarch64-linux
```

Comment:
@idontgetoutmuch has an M1 MBP. With just these he won't be able to build or get into a nix shell.

Reply:
Yeah, the point here was to be explicit about the systems you want to support. Feel free to add more.

Ideally you would also build it on that system.